From 033d5f3cdb8bf1b6a11f382d38e1373db543de8a Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Mon, 08 Apr 2019 06:29:56 +0300
Subject: [PATCH] [SW-0] regs: use specs project macros

do not use registers definition from u-boot but from specs.
This is another step on u-boot decoupling

Change-Id: I70aa94e105a886e2d4eafe22790695810d5fd644
---

diff --git a/common/preboot.c b/common/preboot.c
index 9244add..6825412 100644
--- a/common/preboot.c
+++ b/common/preboot.c
@@ -7,11 +7,15 @@
 
 #include <common.h>
 #include <linux/io.h>
-#include <mach/hl_regs.h>
-#include <mach/hl_fw_if.h>
 #include "auto_generated.h"
 #include <stddef.h>
 #include <stdbool.h>
+
+#include <psoc_global_conf_regs.h>
+#include <hl_fw_if.h>
+#include <hl_boot_if.h>
+
+
 #include "preboot.h"
 #include "uart.h"
 #include "spi.h"
diff --git a/common/preboot.h b/common/preboot.h
index b5b17d0..474c1ad 100644
--- a/common/preboot.h
+++ b/common/preboot.h
@@ -8,28 +8,31 @@
 #ifndef COMMON_PREBOOT_H_
 #define COMMON_PREBOOT_H_
 
+#define HL_SPI_BASE				0x7FF8000000ull
+#define HL_CFG_BASE				0x7FFC000000ull
+
 static inline u8 regmap_readb(u64 offset)
 {
-	u8 reg = *((volatile u8 *)offset);
+	u8 reg = *((volatile u8 *)(HL_CFG_BASE + offset));
 
 	return reg;
 }
 
 static inline void regmap_writeb(u8 val, u64 offset)
 {
-	*((volatile u8 *)offset) = val;
+	*((volatile u8 *)(HL_CFG_BASE + offset)) = val;
 }
 
 static inline u32 regmap_readl(u64 offset)
 {
-	u32 reg = *((volatile u32 *)offset);
+	u32 reg = *((volatile u32 *)(HL_CFG_BASE + offset));
 
 	return reg;
 }
 
 static inline void regmap_writel(u32 val, u64 offset)
 {
-	*((volatile u32 *)offset) = val;
+	*((volatile u32 *)(HL_CFG_BASE + offset)) = val;
 }
 
 u64 get_cpu_boot_status_reg(void);
diff --git a/common/spi.c b/common/spi.c
index 323bac0..7164a6b 100644
--- a/common/spi.c
+++ b/common/spi.c
@@ -9,22 +9,14 @@
 #include <linux/io.h>
 #include <linux/sizes.h>
 #include <linux/bitops.h>
-#include <mach/hl_regs.h>
+
+#include <psoc_gpio3_regs.h>
+#include <psoc_spi_regs.h>
 
 #include "preboot.h"
 #include "spi.h"
 #include "uart.h"
 
-#define mmPSOC_SPI_CTRLR0		(HL_CFG_BASE + 0xC43000)
-#define mmPSOC_SPI_SSIENR		(HL_CFG_BASE + 0xC43008)
-#define mmPSOC_SPI_SER			(HL_CFG_BASE + 0xC43010)
-#define mmPSOC_SPI_BAUDR		(HL_CFG_BASE + 0xC43014)
-#define mmPSOC_SPI_TXFTLR		(HL_CFG_BASE + 0xC43018)
-#define mmPSOC_SPI_RXFTLR		(HL_CFG_BASE + 0xC4301C)
-#define mmPSOC_SPI_TXFLR		(HL_CFG_BASE + 0xC43020)
-#define mmPSOC_SPI_RXFLR		(HL_CFG_BASE + 0xC43024)
-#define mmPSOC_SPI_IMR			(HL_CFG_BASE + 0xC4302C)
-#define mmPSOC_SPI_DR0			(HL_CFG_BASE + 0xC43060)
 #define SPL_SIZE			SZ_64K
 #define CMD_WRITE_STATUS		0x01
 #define CMD_READ			0x03
diff --git a/common/uart.c b/common/uart.c
index 0e89a8c..4cafc5f 100644
--- a/common/uart.c
+++ b/common/uart.c
@@ -6,11 +6,11 @@
  */
 
 #include <linux/io.h>
-#include <mach/hl_regs.h>
+
+#include <psoc_uart_0_regs.h>
+
 #include "preboot.h"
 
-#define mmPSOC_UART_0_DATA		(HL_CFG_BASE + 0xC45000)
-#define mmPSOC_UART_0_STATE		(HL_CFG_BASE + 0xC45004)
 #define UART_SR_TXFULL			(1 << 0) /* TX Buffer Full */
 #define STR_UINT_SZ			8
 
diff --git a/gaudi/gaudi.c b/gaudi/gaudi.c
index 9fe7cbc..28cf7bf 100644
--- a/gaudi/gaudi.c
+++ b/gaudi/gaudi.c
@@ -6,8 +6,9 @@
  */
 
 #include <linux/io.h>
-#include <mach/hl_regs.h>
-#include <mach/../../gaudi/include/mach/gaudi_regs.h>
+
+#include <psoc_global_conf_regs.h>
+#include <pcie_aux_regs.h>
 
 #include "preboot.h"
 
@@ -28,7 +29,6 @@
 	return mmPSOC_GLOBAL_CONF_PAD_SEL_53;
 }
 
-#define mmPSOC_GLOBAL_CONF_CPU_BOOT_STATUS	(HL_CFG_BASE + 0xC4B300)
 u64 get_cpu_boot_status_reg(void)
 {
 	return mmPSOC_GLOBAL_CONF_CPU_BOOT_STATUS;
diff --git a/goya/goya.c b/goya/goya.c
index 363ad29..a2fd6f5 100644
--- a/goya/goya.c
+++ b/goya/goya.c
@@ -7,8 +7,14 @@
 
 #include <common.h>
 #include <linux/io.h>
-#include <mach/hl_regs.h>
-#include <mach/../../goya/include/mach/goya_regs.h>
+
+#include <pcie_phy_regs.h>
+#include <psoc_global_conf_regs.h>
+#include <pcie_aux_regs.h>
+#include <pcie_core_regs.h>
+#include <cpu_timestamp_regs.h>
+#include <pcie_db_rsv_regs.h>
+
 #include "preboot.h"
 
 #define BTL_COUNTER_MAX		BIT(22)
diff --git a/goya/pll.c b/goya/pll.c
index 269e581..3bacfc6 100644
--- a/goya/pll.c
+++ b/goya/pll.c
@@ -29,9 +29,9 @@
 #define BYPASS_CYCLES_PER_USEC	(50000000 / USEC_PER_SEC)
 #define MIN_CYCLES_PER_ITER	5
 
-#define mmCPU_PLL_BASE		0x7FFC4A2000ull
-#define mmIC_PLL_BASE		0x7FFC4A3000ull
-#define mmPSOC_MME_PLL_BASE	0x7FFCC71000ull
+#define mmCPU_PLL_BASE		0x4A2000
+#define mmIC_PLL_BASE		0x4A3000
+#define mmPSOC_MME_PLL_BASE	0xC71000
 
 enum hl_pll_bypass_val {
 	HL_PLL_REF_CLK,
